internal/resources/slo/resource_slo_test.go

```diff
@@ -138,12 +145,19 @@ func testAlertingExists(expectation bool, rn string, slo *gapi.Slo) resource.Tes
 	}
 }
 
-func testAccSloCheckDestroy(slo *gapi.Slo) resource.TestCheckFunc {
+func testAccSloCheckDestroy(slo *slo.Slo) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
-		client := testutils.Provider.Meta().(*common.Client).DeprecatedGrafanaAPI
-		client.DeleteSlo(slo.UUID)
+		client := testutils.Provider.Meta().(*common.Client).SLOClient
+		req := client.DefaultAPI.V1SloIdGet(context.Background(), slo.Uuid)
+		gotSlo, resp, _ := req.Execute()
```

Comment:
Don't you check the error here? If you have any error doing the request maybe you cannot retrieve status code or status type.

Reply:
The resp is always not-nil. And so, to check that a SLO was destroyed, checking that either we get a 404, or an SLO with a "deleting" status is sufficient. All other cases cause a test failure